% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
\chapter{API --- The MG-RAST Application Programming Interface}
\label{API}
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 

% 

% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
\section{URLs}
\begin{small}
\begin{verbatim}
http://api.metagenomics.anl.gov/
\end{verbatim}
\end{small} Further documentation, with a complete parameter listing for all resources available is at:
\begin{small}
\begin{verbatim}
http://api.metagenomics.anl.gov/api.html
\end{verbatim}
\end{small} Github repository of script tools, examples, and contributed code for using the MG-RAST API:
\begin{small}
\begin{verbatim}
https://github.com/MG-RAST/MG-RAST-Tools
\end{verbatim}
\end{small}
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
\section{Introduction}

The API exposes the data and computational results in MG-RAST for consumption by third parties, greatly enhancing the utility of the MG-RAST service.

The RESTful API is a core component of MG-RAST, all future MG-RAST work will be based on the API.


The MG-RAST API enables programmatic access to data and analyses in MG-RAST without requiring local installations. That being said, it requires programming skills. If you are not a programmer or you are not willing to spend the time learning the API, the Example scripts (see chapter \ref{API-Examples}.) 

Using the API, users can authenticate against the service, submit their data, download results, and perform extensive comparisons of data sets. The API uses the Representational State Transfer (REST) [3] architecture which allows download of data in ASCII format, allowing users to query the system via URLs and returning MG-RAST data objects in their native format (e.g. similarity tables or sequence files). For structured data (e.g. metadata or project information) the MG-RAST API uses JSON (Javascript Object Notation, a widely used standard) as its data format. 

This allows users to use simple tools to download data files or view the JSON in their web browsers using one of the many available JSON viewers. In addition many programming languages have libraries for convenient HTTP interaction and JSON conversions. The API has a minimal number of prerequisites; and any language with HTTP and JSON support or command line utilities such as ``curl" can easily integrate with the design. 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
\section{Design and Implementation}
The MG-RAST API enables programmatic access to data and analyses in MG-RAST without requiring local installations. Users can authenticate against the service, submit their data, download results, and perform extensive comparisons of data sets. We chose to use the Representational State Transfer (REST) [3] architecture. The REST approach allows download of data in ASCII format, allowing users to query the system via URLs and returning  MG-RAST data objects in their native format (e.g. similarity tables or sequence files). For structured data (e.g. metadata or project information) the MG-RAST API uses JSON (Javascript Object Notation, a widely used standard) as its data format. 

Using this approach users can use simple tools to download data files to their machines or view the JSON in their web browsers using one of the many available JSON viewers. In addition many programming languages have libraries for convenient HTTP interaction and JSON conversions. 

Most of the API calls are simply URLs which can be entered in the address bar of a web browser to perform the download through the browser. These URLs can also be used with a command line tool like curl, in programing-language-specific libraries, or in command line scripts. The examples in the Results section illustrate the use of each of these methods. The example scripts are available on in the supplementary materials and on github (https://github.com/MG-RAST/MG-RAST-Tools) along with other useful illustrative scripts.

The MG-RAST API covers most of the functionality available through the MG-RAST website, with access to annotations, analyses, metadata and access to the MG-RAST user inbox to view contents as well as upload files. All sequence data and data products from intermediate stages in the analysis pipeline are available for download. Other resources provide services not available through the website, e.g. the m5nr resource lets you query the m5nr database.

Each query to the API is represented as a URI beginning with 
\begin{small}
\begin{verbatim}
http://api.metagenomics.anl.gov/
\end{verbatim}
\end{small} and has a defined structure to pass the requests and parameters to the API server. These URI queries can be used from the command line, e.g. using curl, in a browser, or incorporated in a shell script or program.

Each URI has the form:
\begin{small}
\begin{verbatim}
http://api.metagenomics.anl.gov/{version}/{resourcepath}?{querystring}
\end{verbatim}
\end{small} where

\begin{small}
\begin{verbatim}
{version}
\end{verbatim}
\end{small}
explicitly directs the request to a specific version of the API. If it is omitted the latest API version will be used. The current version number is `1'.

\begin{small}
\begin{verbatim}
{resourcepath}
\end{verbatim}
\end{small}
is constructed from the path parameters listed below to define a specific resource.

\begin{small}
\begin{verbatim}
{querystring}
\end{verbatim}
\end{small}
is used to filter the results obtained for the resource, this is optional.

For example, in:
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt
\end{lstlisting}
\end{small} the resource path 

\begin{small}
\begin{verbatim}
annotation/sequence/mgm4447943.3
\end{verbatim}
\end{small} defines a request for the annotated sequences for the MG-RAST job with ID 4447943.3. 
The optional query string 

\begin{small}
\begin{verbatim}
evalue=10&type=organism&source=SwissProt
\end{verbatim}
\end{small} modifies the results by setting an evalue cutoff, annotation type and database source.

The API provides an authentication mechanism for access to private MG-RAST jobs and users' inbox. The 'auth\_key' (or 'webkey') is a 25 character long string, e.g.

\begin{small}
\begin{verbatim}
j6FNL61ekNarTgqupMma6eMx5
\end{verbatim}
\end{small} which is used by the API to identify an MG-RAST user account and determine access rights to metagenomes. Note that the auth\_key is valid for a limited time after which queries using the key will be rejected. You can create a new auth\_key or view the expiration date and time of an existing auth\_key on the MG-RAST website. An account can have only one valid auth\_key and creating a new key will invalidate an existing key.

All public data in MG-RAST is available without an auth\_key. All API queries for private data which either do not have an auth\_key or use an invalid or expired auth\_key will get a "insufficient permissions to view this data" response.

The auth\_key can be included in the query string like:
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt&auth_key=j6FNL61ekNarTgqupMma6eMx5
\end{lstlisting}
\end{small} or in a request using curl like:
\begin{small}
\begin{lstlisting}
curl -X GET -H "auth: j6FNL61ekNarTgqupMma6eMx5" "http://api.metagenomics.anl.gov/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt"
\end{lstlisting}
\end{small} Note that for the curl command the quotes are necessary for the query to be passed to the API correctly.

If an optional parameter passed through the query string has a list of values only the first will be used. When multiple values are required, e.g. for multiple md5 checksum values, they can be passed to the API like:
\begin{small}
\begin{lstlisting}
curl -X POST -d '{"data":["000821a2e2f63df1a3873e4b280002a8","15bf1950bd9867099e72ea6516e3d602"]}' "http://api.metagenomics.anl.gov//m5nr/md5"
\end{lstlisting}
\end{small}

In some cases, the data requested is in the form of a list with a large number of entries. In these cases the `limit' and `offset' parameters can be used to step through the list, e.g.
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/project?order=name&limit=20&offset=100
\end{lstlisting}
\end{small} will limit the number of entries returned to 20 with an offset of 100. If these parameters are not provided default values of \texttt{limit=10} and \texttt{offset=0} are used. The returned JSON structure will contain the `next' and `prev' (previous) URIs to simplify stepping through the list.

The data returned may be plain text, compressed gzipped files or a JSON structure. 

Most API queries are `synchronous' and results are returned immediately. Some queries may require a substantial time to compute results, in these cases you can select the asynchronous option by adding \texttt{`\&asynchronous=1'} to the end of the query string. This query will then return a URL which will return the query results when they are ready.

Most of the API calls are simply URLs which can be entered in the address bar of a web browser to perform the download through the browser. These URLs can also be used with a command line tool like curl, in programing-language-specific libraries, or in command line scripts. The examples below illustrate the use of each of these methods. The example scripts are available on the github site along with other useful illustrative scripts.

\begin{table}
\caption{Top-level resources available through the MG-RAST-API}
\centering
\renewcommand{\arraystretch}{1.3}
\renewcommand\tabcolsep{10pt}
%\begin{tabular}{ l l }
\begin{tabular}{ p{3cm} p{12cm} }
  \hline
  \hline
  \textbf{Resource/Object} & \textbf{Description} \\
  \hline
   \textbf{annotation} & taxonomic and functional annotations made by comparison with the M5nr database \\
   \textbf{compute} & resource to compute PCoA , heatmap, and normalization for a set of input metagenomes \\
   \textbf{download} & download results of the MG-RAST pipeline \\
   \textbf{inbox} & upload and listing of data in the staging area prior to execution of the MG-RAST pipeline \\
   \textbf{library} & library information for uploaded metagenome provided by the user \\
   \textbf{matrix} & abundance profiles in BIOM (5) format for a list of metagenomes \\
   \textbf{M5nr} & access M5 nonredundant protein database used for annotation of metagenomic sequences \\
   \textbf{metadata} & creation, export, and validation of metadata templates and spreadsheets \\
   \textbf{metagenome} & container for sample, library, project, and precomputed data for an uploaded metagenomic sequence file \\
   \textbf{profile} & returns a single data object in BIOM format \\
   \textbf{project} & project summary for metagenome provided by user \\
   \textbf{sample} & sample information provided by user \\
   \textbf{search} & search MG-RAST by MG-ID, metadata, function, or taxonomy; or implement a more complex search. \\
   \textbf{validation} & validates templates for correct structure and data \\
  \hline
  \hline
\end{tabular}
\label{table:upload_speeds}
\end{table}
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
\section{Examples}
The API provides index-driven access to data subsets using the following data types as indices into the data: functions, functional hierarchy data, and taxonomic data. Whenever possible we have employed standards to expose data and metadata, such as the BIOM standard for encoding abundance profiles. The examples below are intended to illustrate usage for the various resources available, they do not cover the entire functionality of the API, see the documentation at the API website for the comprehensive listing.

\begin{itemize}
\item 
\textbf{annotation}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/annotation/sequence/mgm4440036.3?type=function&filter=protease&source=Subsystems
\end{lstlisting}
\end{small} Retrieve the reads from a metagenome with ID mgm4440036.3 which were annotated as  protease in SEED Subsystems.

\item
\textbf{download}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/download/mgm4447943.3
\end{lstlisting}
\end{small} Retrieve information formatted as a JSON object about all the files available for download for metagenome mgm4447943.3 with information about the files and sequence statistics where applicable. Each file listed has a URL included which can be used to download the file, e.g.
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/download/mgm4447943.3?file=650.1
\end{lstlisting}
\end{small} will download the protein.sims file containing the BLAT similarities.

\item
\textbf{inbox}
\begin{small}
\begin{lstlisting}
curl -X POST -H "auth: auth_key" -F "upload=@sequences.fastq" "http://api.metagenomics.anl.gov/1/inbox"
\end{lstlisting}
\end{small}
Upload the file 'sequences.fastq' to your inbox. This API call requires user authentication using the auth\_key described above. It can not be used in a browser, but needs to be run from the command line or from a script.

\item
\textbf{matrix}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/matrix/organism?group_level=family&source=SEED&evalue=5&id=mgm4440442.5&id=mgm4440026.3
\end{lstlisting}
\end{small}
Retrieve the taxonomic abundance profile on family level for 2 metagenomes based on SEED assignments with an evalue cutoff of 1e-5.

\item
\textbf{metagenome}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/metagenome/mgm4440026.3
\end{lstlisting}
\end{small} List analysis submission parameters and other details for a metagenome.
\newline
The metagenome resource can also be used to search metadata, function and taxonomy.  
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/metagenome?function=dnaA&organism=coli&biome=marine&match=all&order=created
\end{lstlisting}
\end{small} This call will find all marine metagenomes with reads annotated as dnaA and have taxonomic assignment containing the text `coli', the results will be ordered based on creation date for the metagenome.

\item
\textbf{project}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/project/mgp31?verbosity=full
\end{lstlisting}
\end{small} Retrieve available information about the project with ID mgp31.

\item
\textbf{sample}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/sample/mgs12326?verbosity=full
\end{lstlisting}
\end{small} Retrieve available information about individual samples, including IDs and metadata.

\item
\textbf{metadata}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov//metadata/template
\end{lstlisting}
\end{small} Retrieve the static template for metadata object relationships and types used by MG-RAST.
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov//metadata/export/mgp128
\end{lstlisting}
\end{small} Retrieve all metadata for project mgp128.
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/metadata/cv
\end{lstlisting}
\end{small} Retrieve a set of lists of all our controlled metadata terms, including the ontologies.
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/metadata/ontology?name=biome&version=2013-04-27
\end{lstlisting}
\end{small} Retrieve a more detailed list (with relationships) for a specific version of the ontology.

\item
\textbf{m5nr}
\begin{small}
\begin{lstlisting}
http://api.metagenomics.anl.gov/1/m5nr/md5/ffc62262a18b38671c3e337150ef535f?source=SwissProt
\end{lstlisting}
\end{small} Retrieve the UniProt ID for a given sequence identifier.

\end{itemize}
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
