
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>FAQ – Frequently asked questions about MG-RAST &#8212; MG-RAST  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="API — The MG-RAST Application Programming Interface" href="api.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="faq-frequently-asked-questions-about-mg-rast">
<h1>FAQ – Frequently asked questions about MG-RAST<a class="headerlink" href="#faq-frequently-asked-questions-about-mg-rast" title="Permalink to this headline">¶</a></h1>
<p>The answers to some of these Frequently Asked Questions can be found
elsewhere in this manual, they are listed here for users who would like
a quick answer to a simple question. Other sections of the manual will
generally contain more detail than the answers in this chapter. Some
answers are just links to relevant sections in other chapters.</p>
<div class="section" id="general">
<h2>General<a class="headerlink" href="#general" title="Permalink to this headline">¶</a></h2>
<div class="section" id="what-is-mg-rast">
<h3>What is MG-RAST?<a class="headerlink" href="#what-is-mg-rast" title="Permalink to this headline">¶</a></h3>
<p>The MG-RAST server is an open source system for annotation and
comparative analysis of metagenomes. Users can upload raw sequence data
in fasta format; the sequences will be normalized and processed and
summaries automatically generated. The server provides several methods
to access the different data types, including phylogenetic and metabolic
reconstructions, and the ability to compare the metabolism and
annotations of one or more metagenomes and genomes. In addition, the
server offers a comprehensive search capability. Access to the data is
password protected, and all data generated by the automated pipeline is
available for download in a variety of common formats.</p>
</div>
<div class="section" id="contacting-the-mg-rast-team-and-help-desk">
<span id="section-contact-mgrast"></span><h3>Contacting the MG-RAST team and help desk<a class="headerlink" href="#contacting-the-mg-rast-team-and-help-desk" title="Permalink to this headline">¶</a></h3>
<p>The MG-RAST project uses a ticket system to manage interactions with
users, please use the email address for the MG-RAST project shown in
Figure <a class="reference external" href="#fig:mgrastemail">9.1</a>.</p>
<div class="figure" id="fig-mgrastemail">
<a class="reference internal image-reference" href="_images/mgrastemail.png"><img alt="The email address for the MG-RAST project. Note that it was inserted into this document as an image and can not be copied as text, you will have to type it." src="_images/mgrastemail.png" style="width: 3in;" /></a>
<p class="caption"><span class="caption-text">The email address for the MG-RAST project. Note that it was inserted
into this document as an image and can not be copied as text, you
will have to type it.</span></p>
</div>
<p>We recommend including as much detail as possible into your emails to
the help-desk, details like account names, MG-RAST identifiers will help
us identify any issues and speed up resolving them.</p>
<p>Below are examples of the types of details we would like to receive:</p>
<ul class="simple">
<li>your name</li>
<li>your account name for MG-RAST (please do NOT include your password or
webkey)</li>
<li>a clear text description of your problem</li>
<li>any MG-RAST identifiers (those are the 444xxxx.3 numbers)</li>
<li>any project numbers</li>
<li>the browser and which version you are using, if the problem relates
to the web site</li>
<li>what platform your data was created on</li>
<li>if your data was a failure in the web site, what time the failure
occurred</li>
<li>the URL and name of the page you were viewing</li>
<li>screenshot(s) of the problem</li>
</ul>
</div>
<div class="section" id="what-kinds-of-data-sets-does-mg-rast-analyze">
<h3>What kinds of data sets does MG-RAST analyze?<a class="headerlink" href="#what-kinds-of-data-sets-does-mg-rast-analyze" title="Permalink to this headline">¶</a></h3>
<p>MG-RAST is designed to annotate a large set of nucleotide sequences, not
a complete genome and not amino acid sequences. The RAST server should
be used if you want to annotate complete, or nearly complete prokaryotic
genomes. Version 3.2 accepts reads of length 75bp and up, and is capable
of handling sequences of several dozen kilobases. For whole metagenome
shotgun data we use a gene prediction step that is not suitable for
eukaryotes, for that reason do not expect MG-RAST v3.2 to work with
eukaryotic data sets or for the eukaryotic subsets of your data.</p>
</div>
<div class="section" id="how-many-metagenomes-can-i-submit">
<h3>How many metagenomes can I submit?<a class="headerlink" href="#how-many-metagenomes-can-i-submit" title="Permalink to this headline">¶</a></h3>
<p>We do not restrict user submission of samples. However, the computation
required is massive and samples are processed on a first-come,
first-serve basis. MG-RAST v3 is over 200 times faster than the previous
version. We will also provide a CLOUD client (shortly after the initial
release) that connects to MG-RAST and will allow you to add processing
power to your jobs in MG-RAST.</p>
</div>
<div class="section" id="can-i-use-mg-rast-as-a-repository-for-my-metagenomic-data">
<h3>Can I use MG-RAST as a repository for my metagenomic data?<a class="headerlink" href="#can-i-use-mg-rast-as-a-repository-for-my-metagenomic-data" title="Permalink to this headline">¶</a></h3>
<p>MG-RAST has become an unofficial repository for metagenomic data,
providing a means to make your data public so that it is available for
download and viewing of the analysis without registration, as well as a
static link that you can use in publications. It also requires that you
include experimental metadata about your sample when it is made public
to increase the usefulness to the community. We undertake to maintain
public datasets within MG-RAST and they are not subject to deletion.</p>
</div>
<div class="section" id="who-should-i-contact-with-questions-or-problems-with-mg-rast">
<h3>Who should I contact with questions or problems with MG-RAST?<a class="headerlink" href="#who-should-i-contact-with-questions-or-problems-with-mg-rast" title="Permalink to this headline">¶</a></h3>
<p>All questions, comments or problems regarding MG-RAST should be directed
to our support team using either the letter symbol in the navigation
toolbox or via email to:</p>
<p><code class="docutils literal notranslate"><span class="pre">help</span> <span class="pre">at</span> <span class="pre">mg-rast.org</span></code></p>
<p>.</p>
</div>
<div class="section" id="how-should-i-link-to-mg-rast-in-a-publication">
<h3>How should I link to MG-RAST in a publication?<a class="headerlink" href="#how-should-i-link-to-mg-rast-in-a-publication" title="Permalink to this headline">¶</a></h3>
<p>You can provide a stable link to an MG-RAST job or project using the
following URLs:</p>
<p><code class="docutils literal notranslate"><span class="pre">http://mg-rast.org/linkin.cgi?metagenome=</span></code></p>
<p><code class="docutils literal notranslate"><span class="pre">http://mg-rast.org/linkin.cgi?project=</span></code></p>
<p>For example, for the metagenome ID 4440283.3 the URL is:</p>
<p><a class="reference external" href="http://mg-rast.org/linkin.cgi?metagenome=4440283.3">http://mg-rast.org/linkin.cgi?metagenome=4440283.3</a></p>
<p>This URL provides a stable method of linking to your data which does not
require the viewer to have an MG-RAST account. Please do not use the URL
you see when you are browsing the site.</p>
<p>Note that by default your data is not visible to others, you will need
to explicitly grant permission for it to be visible to anyone on the
internet by making it public through the MG-RAST website.</p>
</div>
<div class="section" id="section-identifier">
<span id="identifiers"></span><h3>Identifiers<a class="headerlink" href="#section-identifier" title="Permalink to this headline">¶</a></h3>
<p>MG-RAST automatically assigns a unique identifier to every dataset
submitted. Upon completion of the automated pipeline, datasets can be
viewed via the web interface by using the identifiers. The dataset
identifiers are of the form <code class="docutils literal notranslate"><span class="pre">integer</span> <span class="pre">prefix</span></code>.<code class="docutils literal notranslate"><span class="pre">revision</span></code>. An
example is <code class="docutils literal notranslate"><span class="pre">4440283.3</span></code>.</p>
<p>In addition to individual datasets, projects (groups of datasets) can be
addressed with simple numerical project identifiers. An example is
<code class="docutils literal notranslate"><span class="pre">128</span></code>.</p>
</div>
<div class="section" id="linking-to-mg-rast">
<span id="section-linkin"></span><h3>Linking to MG-RAST<a class="headerlink" href="#linking-to-mg-rast" title="Permalink to this headline">¶</a></h3>
<p>Because future versions of MG-RAST may change, we provide a link-in
mechanism as a stable way of linking to MG-RAST. To link to datasets or
projects in MG-RAST, users should always use the <code class="docutils literal notranslate"><span class="pre">linkin.cgi</span></code>,
especially in publications.</p>
<p>Note: You must make the data set PUBLIC before you can publicly share
the link. It will not work for others until you do.</p>
<p>Note: Do not use the URL that is displayed in the browser when browsing
the site.</p>
<p><code class="docutils literal notranslate"><span class="pre">https://mg-rast.org/linkin.cgi?metagenome=</span></code></p>
<p><code class="docutils literal notranslate"><span class="pre">https://mg-rast.org/linkin.cgi?project=</span></code></p>
<p>For example, for the public dataset with metagenome ID <code class="docutils literal notranslate"><span class="pre">4440283.3</span></code> the
URL is: <a class="reference external" href="http://mg-rast.org/linkin.cgi?metagenome=4440283.3">http://mg-rast.org/linkin.cgi?metagenome=4440283.3</a>. For the
public project with project ID <code class="docutils literal notranslate"><span class="pre">128</span></code> the URL is:
<a class="reference external" href="http://mg-rast.org/linkin.cgi?project=128">http://mg-rast.org/linkin.cgi?project=128</a>.</p>
<p>These URLs provides a stable method of linking to data that does not
require the viewer to have an MG-RAST account.</p>
</div>
<div class="section" id="privacy">
<span id="section-data-visibility"></span><h3>Privacy<a class="headerlink" href="#privacy" title="Permalink to this headline">¶</a></h3>
<p>By default, a user’s data is not visible to others; the user needs to
explicitly grant permission for the data to be visible to anyone on the
Internet, either by sharing with individuals or by making it public
through the MG-RAST website. Only the owner of a dataset (the original
submitter) can make a dataset public and this requires explicit action
on their part, MG-RAST does not make data public without this action.
Owners can grant anonymous access to manuscript reviewers (see Section
<a class="reference external" href="#section:reviewer_sharing">[section:reviewer_sharing]</a>).</p>
<p>The web interface allows sharing and publication of data, requiring the
presence of minimal metadata (see Section <a class="reference external" href="#section:metadata">4.7</a>)
for data that is made public. Data can be shared or made public only
after the computation has finished.</p>
<div class="section" id="sharing-with-individual-users">
<span id="section-user-sharing"></span><h4>Sharing with individual users<a class="headerlink" href="#sharing-with-individual-users" title="Permalink to this headline">¶</a></h4>
<p>Data and analyses can be shared with individual users. To share data,
users simply enter their email address via clicking the <code class="docutils literal notranslate"><span class="pre">Sharing</span></code> link
on the Metagenome Overview page. The dialogue shown in Figure
<a class="reference external" href="#fig:sharing">9.2</a> will allow entering email addresses.</p>
<div class="figure" id="fig-sharing">
<a class="reference internal image-reference" href="Images/sharing"><img alt="Dialogue showing the sharing mechanism. The mechanism requires a valid email address for the user with whom the data is to be shared. A list of users with access to the data is displayed at the bottom on the page." src="Images/sharing" style="width: 6in;" /></a>
<p class="caption"><span class="caption-text">Dialogue showing the sharing mechanism. The mechanism requires a
valid email address for the user with whom the data is to be shared.
A list of users with access to the data is displayed at the bottom on
the page.</span></p>
</div>
<p>Both individual jobs as well as entire projects containing one or more
jobs can be shared using a similar mechanism from the Job Overview and
Project pages respectively.</p>
<p>As shown in Figure <a class="reference external" href="#fig:Data-sharing-in-mg-rast">9.3</a>, we tend to
see dataset sharing between small groups of users.</p>
<div class="figure" id="fig-data-sharing-in-mg-rast">
<a class="reference internal image-reference" href="Images/Data-sharing-in-mg-rast"><img alt="Data sets shared in MG-RAST by users (orange dots), shown as connecting edges." src="Images/Data-sharing-in-mg-rast" style="width: 3in;" /></a>
<p class="caption"><span class="caption-text">Data sets shared in MG-RAST by users (orange dots), shown as
connecting edges.</span></p>
</div>
</div>
<div class="section" id="anonymous-sharing-with-reviewers">
<span id="section-reviewer-sharing"></span><h4>Anonymous sharing with reviewers<a class="headerlink" href="#anonymous-sharing-with-reviewers" title="Permalink to this headline">¶</a></h4>
<p>To grant manuscript reviewers access to a project while preserving their
anonymity click on the ’Create Reviewer Access Token’ button on the
project page. This button is visible only to the owner of a project by
clicking on the ’Share Project’ link. It will generate a token that can
be sent to the publisher to pass on to reviewers. When a reviewer
receives the token from the publisher they need to use the included link
to access MG-RAST. If necessary the reviewer will need to register for
an account and their account will have anonymous access to the project.
The number of reviewers who have accessed the project is displayed to
the owner in the list of users the project is shared with, but the
identity of the reviewers is not disclosed. The owner of the project can
revoke the token at any time to disable access.</p>
</div>
</div>
<div class="section" id="publishing">
<span id="section-publishing"></span><h3>Publishing<a class="headerlink" href="#publishing" title="Permalink to this headline">¶</a></h3>
<p>MG-RAST provides a mechanism to make data and analyses publicly
accessible. All sequence data, metadata, analyses, and analyses files
for a dataset will be freely available for download once it is made
public. Only the submitting user can make data public on MG-RAST and
once this is done it can not be reversed. Metadata is mandatory for
dataset publication (see Section <a class="reference external" href="#section:metadata">4.7</a>).</p>
<p>The following checklist describes the process of making MG-RAST datasets
and projects public:</p>
<ol class="arabic">
<li><p class="first">Ownership of the datasets: To make a dataset public your account
needs to be labelled as the owner in MG-RAST.</p>
</li>
<li><p class="first">Ownership of the project: Your account should be the owner of the
project as well, this is usually just the account that was used to
create the project.</p>
</li>
<li><p class="first">Metadata: MG-RAST requires that you enter metadata for the project,
samples and libraries before it is made public to increase its
utility to the community. This is done through a pre-formatted excel
spreadsheet which you fill in with the necessary metadata. If you
have already entered metadata, e.g. during submission, and want to
make changes, you can download this file with the existing metadata
prefilled from the project page with the ‘Export Metadata’ link.</p>
<p>If you have not entered metadata for your project, download the
latest metadata template file from:
<a class="reference external" href="ftp://ftp.mg-rast.org/data/misc/metadata/">ftp://ftp.mg-rast.org/data/misc/metadata/</a> The first sheet is a README
containing some important tips for entering the metadata. The second
row in each sheet in the template contains some explanation and
instructions for each column. The columns marked with red headers are
required.</p>
<p>You can enter your data directly into the template, a better route
would be to use the tool we built to facilitate metadata entry –
MetaZen: <a class="reference external" href="http://mg-rast.org/metazen.cgi">http://mg-rast.org/metazen.cgi</a>. MetaZen will step you
through the data entry and then give you a pre-filled excel
spreadsheet to download which you can then edit further if necessary.</p>
<p>Once you have the metadata file ready, upload with the ‘Upload
Metadata’ link on the project page.</p>
</li>
<li><p class="first">Release metagenomes: Make each dataset public, there is a ‘Make
public’ link in the blue bar near the top of the Metagenome Overview
page.</p>
</li>
<li><p class="first">Project Data: Edit the project page information if you wish with the
‘Edit Project Data’ link. You can enter an abstract, links to
publications, additional description, contacts etc. This page is the
central point in MG-RAST from where people will access your data and
analyses so add all information that may be useful.</p>
</li>
<li><p class="first">Final step: Make the project public from the project page (project
page blue bar, ‘Make Public’).</p>
</li>
</ol>
<p>The link for a public project which should be used in a publication is
listed near the top of the project page, e.g.:
<a class="reference external" href="http://mg-rast.org/metagenomics.cgi?page=MetagenomeProject&amp;project=128">http://mg-rast.org/metagenomics.cgi?page=MetagenomeProject&amp;project=128</a>
where 128 is the MG-RAST project ID.</p>
<p>The link for individual public metagenomes which should be used in a
publication is listed near the top of the metagenome overview page,
e.g.:
<a class="reference external" href="http://mg-rast.org/linkin.cgi?metagenome=4440283.3">http: //mg-rast.org/linkin.cgi?metagenome=4440283.3</a>
where 4440283.3 is the MG-RAST metagenome ID.</p>
<p>The publication to cite for MG-RAST is at
<a class="reference external" href="http://www.biomedcentral.com/1471-2105/9/386">http://www.biomedcentral.com/1471-2105/9/386</a>.</p>
</div>
<div class="section" id="who-should-i-cite-when-i-use-this-service">
<h3>Who should I cite when I use this service?<a class="headerlink" href="#who-should-i-cite-when-i-use-this-service" title="Permalink to this headline">¶</a></h3>
<p>See Section <a class="reference external" href="#section:MG-RAST-citation">1.5</a>.</p>
</div>
<div class="section" id="is-mg-rast-open-source-and-can-i-install-it-locally">
<h3>Is MG-RAST open source and can I install it locally?<a class="headerlink" href="#is-mg-rast-open-source-and-can-i-install-it-locally" title="Permalink to this headline">¶</a></h3>
<p>MG-RAST is indeed open source. We make the current stable versions
available on github: <a class="reference external" href="https://github.com/MG-RAST/">https://github.com/MG-RAST/</a> However MG-RAST is a
complex system to install (note: we have not been funded to create a
readily installable version) and even more complex to operate. We advise
against attempting to create a private installation and can not provide
any help installing MG-RAST locally.</p>
<p>If you are a biologist worried about runtime of your jobs, there is a
way to run your jobs on computational resources provided by you that
will significantly help. Please contact us at our usual address mg-rast
at mg-rast.org to inquire about ways of setting this up.</p>
<p>If you are a bioinformatician and want to contribute code or test
alternatives for individual steps, we are currently preparing a system
that will make all components of MG-RAST easily accessible. This is not
currently sea-worthy. Same as with the biologists, please contact us at
help at mg-rast.org for details.</p>
</div>
</div>
<div class="section" id="accounts">
<h2>Accounts<a class="headerlink" href="#accounts" title="Permalink to this headline">¶</a></h2>
<p>The analyses of all public datasets in MG-RAST can be viewed in entirety
without an MG-RAST account. An account is required to submit sequence
data for analysis or view the analyses of datasets which have been
shared with you.</p>
<p><strong>Accounts are for individuals, not services or groups.</strong> In our
experience account sharing (e.g. two or more users having access to the
same username/password information) will always lead to problems, we
<strong>strongly</strong> discourage account sharing.</p>
<p>As scientist typically will switch employers every few years we
encourage users to provide two email addresses, the primary email
address could be your work email, the secondary your private email. By
providing a second email address you can avoid losing access to your
account if and when you switch employers and your work email is no
longer available.</p>
<div class="section" id="account-registration">
<h3>Account registration<a class="headerlink" href="#account-registration" title="Permalink to this headline">¶</a></h3>
<p>Use the “Register” link on the front page of the website to request an
account with MG-RAST, you will need to enter a unique login name and
email address along with other minimal information. Use an email address
you use regularly as it will be used to communicate with you when
necessary. After registering you will receive an automated email with a
temporary password after your account has been authorized, usually
within a day.</p>
<p>If you forget your password you can request a new password on the
MG-RAST website using your login and registered email address, a new
password will be generated and sent by email to this address.</p>
</div>
<div class="section" id="account-webkey">
<span id="section-webkey"></span><h3>Account webkey<a class="headerlink" href="#account-webkey" title="Permalink to this headline">¶</a></h3>
<p>The webkey is a unique string of text, e.g. “b8Dvg2d5DCp7KsWKBPzY2GS4i”
associated with your account which is used by MG-RAST for identification
purposes. Your webkey is valid for a limited time period after which it
expires and will not work anymore. You can generate a new webkey at any
time, even if your current webkey has not expired.</p>
<p>The MG-RAST website provides two locations where you can generate a new
webkey:</p>
<ol class="arabic simple">
<li>Log in to MG-RAST and go to the Account Management page. Press the
button under “Preferences” to go the the Manage Preferences page
where the Web Services section displays your current webkey with its
termination date. Click on the “generate new key” button to generate
a new key and then click the “set preferences” button.</li>
<li>Log in to MG-RAST and go to the Upload page and click on the
“generate webkey” button in the “upload files” tab and then click on
the “generate new key” button.</li>
</ol>
<p>Note that generating a new webkey will invalidate your old webkey and
your new webkey will be valid until the termination date displayed on
the page.</p>
</div>
<div class="section" id="why-do-i-need-to-register-for-this-service">
<h3>Why do I need to register for this service?<a class="headerlink" href="#why-do-i-need-to-register-for-this-service" title="Permalink to this headline">¶</a></h3>
<p>If you do not plan to submit data for analysis to MG-RAST and only want
to browse data which is publicly available there is no need to register.
Otherwise we request that users register, with a valid email address, so
we can contact you once the computation is finished or in case user
intervention is required.</p>
</div>
<div class="section" id="i-have-forgotten-my-password-what-should-i-do">
<h3>I have forgotten my password, what should I do?<a class="headerlink" href="#i-have-forgotten-my-password-what-should-i-do" title="Permalink to this headline">¶</a></h3>
<p>In the navigation toolbox (top right corner of the webpage) there is a
’Forgot?’ link displayed. Click on this and enter your login and the
email address you registered with MG-RAST. A changed password will be
sent by email to this address. For security purposes you should login
and change this new password as soon as you receive the email.</p>
</div>
<div class="section" id="can-i-change-my-account-information">
<h3>Can I change my account information?<a class="headerlink" href="#can-i-change-my-account-information" title="Permalink to this headline">¶</a></h3>
<p>Yes, you can change or modify your password, email address, name and
funding source for your account. Login and make the changes on the
account management page.</p>
</div>
</div>
<div class="section" id="upload-and-submission">
<span id="section-uploading-to-mg-rast"></span><h2>Upload and Submission<a class="headerlink" href="#upload-and-submission" title="Permalink to this headline">¶</a></h2>
<p>MG-RAST was designed to allow users to upload sequence data directly
from next-generation sequencing machines. Data can be in FASTA, FASTQ,
or SFF format.</p>
<p>We suggest uploading raw data (in FASTQ or SFF format) and letting
MG-RAST perform the quality control step because this approach will
allow us to identify any issues with the sequencing run. Frequently,
local quality control will identify some issues but mask others.</p>
<p>Compressing large files will reduce the upload time and the chances of a
failed upload. Users can upload gzip (.gz) and bzip2 (.bz2) or Zip
(.zip) files, as well as tar archives compressed with gzip (.tar.gz) or
bzip2 (.tar.bz2).</p>
<p>It is not necessary to assemble data prior to upload to MG-RAST. The
system has been optimized for short reads and can handle uploads of many
hundreds of gigabytes.</p>
<p>Assembled data can be uploaded to MG-RAST and read abundance information
for contigs can be imported as well from FASTA files. The “assembled”
option for the pipeline will attempt to retrieve read abundance
information from the FASTA sequence files.</p>
<div class="section" id="data-submission-via-the-web-interface">
<h3>Data submission via the web interface<a class="headerlink" href="#data-submission-via-the-web-interface" title="Permalink to this headline">¶</a></h3>
<p>To start uploading data to MG-RAST through the website, click on the
green up arrow. Doing so opens the Upload page. On this page you can
upload files, modify the files where needed, add metadata, and submit
files for analysis.</p>
<p>The page has three stages (see Figure
<a class="reference external" href="#fig:submission_stages">[fig:submission_stages]</a>). The first
“Upload” to upload, manipulate, and collect all the files required for a
submission, and “Submit,” to create the MG-RAST job(s), set analysis
parameters, and start the analysis. The last is “Progress”, where you
can monitor your job status.</p>
<div class="figure" id="id2">
<a class="reference internal image-reference" href="_images/submission_stages.png"><img alt="The flow for MG-RAST submissions via the web interface" src="_images/submission_stages.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">The flow for MG-RAST submissions via the web interface</span></p>
</div>
<p>[fig:submission_stages]</p>
<div class="figure" id="id3">
<a class="reference internal image-reference" href="_images/upload_button.png"><img alt="The MG-RAST upload page with its three main stages" src="_images/upload_button.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">The MG-RAST upload page with its three main stages</span></p>
</div>
<p>[fig:upload_button]</p>
<p>Starting with version 3.6 of MG-RAST, the web upload page will provide
significantly more user guidance and help with ensuring the files
uploaded are both compliant with the required naming scheme and are
transferred intact.</p>
<div class="section" id="data-requirements-for-upload">
<h4>Data requirements for upload<a class="headerlink" href="#data-requirements-for-upload" title="Permalink to this headline">¶</a></h4>
<p>Files larger than 50 MB should be compressed before upload, using gzip
(preferable), bzip2 or Zip (less than 4 GB in size). Compression will
reduce the time taken for the upload of the file, which in turn reduces
the chance that the upload will fail. The requirements for submission
are sequence information (required), metadata (strongly recommended) and
barcode information (for multiplexed datasets only).</p>
<p>We note that priority will be giving to data that has compete GSC
metadata and has been marked for eventual release to the public. The
data release is under user control, MG-RAST staff will not release the
data for the user.</p>
<p>To ensure files are uploaded properly, MG-RAST performs automatic
MD5 <a href="#id10"><span class="problematic" id="id1">[4]_</span></a> checking on client and server side (for most files) to ensure
that files are received correctly by MG-RAST. This is an important part
of data hygiene as files may get corrupted in flight. The new interface
(from version 3.6 onwards), will check the integrity and will give you
immediate feedback about whether your upload was successful. If not
detected at upload time, a damaged file will lead to errors later in the
pipeline, wasting both valuable compute cycles and, even more
importantly, your time.</p>
<p>All files uploaded to MG-RAST should be named using only alphanumeric
and ._ characters without spaces. As of version 3.6, the upload system
ensures that files are compliant with the mandatory naming scheme, using
only alphanumeric and .-characters without spaces. In addition, there is
no need to extract/uncompress files after upload. MG-RAST does this
automatically along with checking metadata and sequence file format and
nomenclature compliance.</p>
<p>Advanced options provides the option to change chunk size. Chunked
uploading allows us to break a large file into small chunks, and send
these pieces to the upload server one-by-one. If an upload fails, we
need only resume from the last successful chunk and allows for resuming
uploads. As a rule, the larger the file and the faster your connection,
the larger the chunk size should be. Set the size lower if your
connection is slow. We have a default setting that works well for most
data sets and connection speeds. If you are encountering upload failure
(outside of formatting issues), try a smaller chunk size.</p>
<p>The following three kinds of files can be uploaded:</p>
<ul>
<li><p class="first">Sequence files</p>
<p>Sequence files must be in either FASTA, FASTQ, or SFF formats</p>
<p>Sequence file names must have one of the following extensions –
‘.fasta’, ‘.fna’, ‘.fastq’, ‘.fq’, or ‘.sff’.</p>
<p>FASTA and FASTQ files should be in plain text ASCII.</p>
<p>FASTA files (and all sequence data submitted to MG-RAST) should not
contain protein sequences.</p>
<p>Assembled data with read abundance information must be in FASTA
format and the coverage included in the sequence ID using the
following simple format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">sequence_number_1_</span><span class="p">[</span><span class="n">cov</span><span class="o">=</span><span class="mi">2</span><span class="p">]</span>
<span class="n">CTAGCGCACATAGCATTCAGCGTAGCAGTCACTAGTACGTAGTACGTACC</span>
<span class="o">&gt;</span><span class="n">sequence_number_2_</span><span class="p">[</span><span class="n">cov</span><span class="o">=</span><span class="mi">4</span><span class="p">]</span>
<span class="n">ACGTAGCTCACTCCAGTAGCAGGTACGTCGAGAAGACGTCTAGTCATCAT</span>
<span class="o">....</span>
</pre></div>
</div>
<p>The abundance information must be appended without spaces to the end
of the sequence name (also without whitespace) in the format
“_[cov=n]”, where n is the coverage or abundance of each contig.
Sequence files in this format should be submitted with the
“assembled” option selected and the pipeline will retrieve read
abundance information from the sequence file.</p>
</li>
<li><p class="first">Metadata file</p>
<p>We provide a spreadsheet template that can be filled out with all the
available metadata information for a dataset, there is a link to the
template on the upload page. Download the template and edit to
include as much information as is available. While the number of
fields in the template is large, the number of required fields,
colored in red in the template, is small. The template file can be
used to upload metadata for one or multiple samples and submit them
to MG-RAST as a single project. The metadata can be modified at any
time after submission to add information or to correct errors. See
Section
<a class="reference external" href="#section:generating_metadata">[section:generating_metadata]</a> for
more details.</p>
<p>We note that a good strategy is to copy an existing metadata file and
modify the values appropriately. Our experience has also shown that
editing the metadata file with tools other than Microsoft Excel will
corrupt the files.</p>
</li>
<li><p class="first">Barcode file</p>
<p>Barcoding reads allows multiplexing multiple samples into a single
sequence file. Barcode files allow demultiplexing those files.
Consequently, Barcode files are required only for sequence data which
will be demultiplexed on the MG-RAST website. In many cases
(typically for shotgun metagenomes) the demultiplexing will have
already been done by the sequencing center. If you have demultiplexed
sequence data, you do not need to enter the barcodes associated with
your samples in a Barcode file. While suitable for all kinds of
barcodes and sequence data, we expect the built-in demultiplexing to
be used mostly for custom barcoded amplicon sequences.</p>
<p>The barcode file should be in plain text ASCII.</p>
<p>If the sequencing facility generated the libraries and did not
demultiplex them for you, make sure to get the barcodes corresponding
to each of your samples. The barcode file should be in plain text
ASCII, a downloadable example can be found at:
<a class="reference external" href="ftp://ftp.mg-rast.org/data/manual/example/">ftp://ftp.mg-rast.org/data/manual/example/</a>.</p>
<p>Each line of the file should contain a single barcode sequence
followed by a tab and then a unique filename, with as many lines as
necessary for the barcodes in the sequence file you are submitting.
Additional columns are ignored.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Example</span><span class="p">:</span>
<span class="n">ACTCTCGTG</span>    <span class="n">sample_1</span>
<span class="n">CAGACATCT</span>    <span class="n">sample_2</span>
<span class="n">GTAGATCAC</span>    <span class="n">sample_3</span>
</pre></div>
</div>
<p>The barcode file typically will be provided by whoever created the
amplicons, in many cases that is the sequencing center.</p>
</li>
</ul>
</div>
<div class="section" id="uploading-data">
<h4>Uploading data<a class="headerlink" href="#uploading-data" title="Permalink to this headline">¶</a></h4>
<p>In this first step, data is uploaded into your private inbox on the
MG-RAST server, this area is write-only and only accessible to you. Data
in the inbox cannot be read or re-exported, its sole purpose is to serve
as a starting point for the pipeline.</p>
<p>When an upload is started it can be aborted or paused. Pausing will
cause the current chunk to complete and then pause the upload. Abort
will interrupt the upload immediately. A paused upload can be resumed by
clicking the resume button in the upload dialog. Aborted uploads can be
resumed or deleted just like other incomplete uploads by clicking the
resume button in the top bar.</p>
<p>When an upload completes (that is not an archived file), an automatic
md5 check will be calculated and the result presented to the user. In
the case of an archive file uploaded, the user has to produce the
checksum of the local file themselves and can paste it into a check
field for validation. A note will be displayed to the user to calculate
the md5sum on the uncompressed file. Archived files will be decompressed
automatically.</p>
<p>At upload:</p>
<ul class="simple">
<li>Sequence files will automatically trigger sequence stats calculation</li>
<li>Sequence files with calculated stats will display those stats upon
selection</li>
<li>Sequence files will show buttons for demultiplexing and joining of
paired ends</li>
<li>Barcode files will automatically show a button for demultiplexing</li>
</ul>
<div class="figure" id="id4">
<a class="reference internal image-reference" href="_images/upload_inbox.png"><img alt="The main elements of the file browser explained. The left side pane shows a list of uploaded files. The top bar provides available actions. Users can select files to view information and whether the file passes formatting check." src="_images/upload_inbox.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">The main elements of the file browser explained. The left side pane
shows a list of uploaded files. The top bar provides available
actions. Users can select files to view information and whether the
file passes formatting check.</span></p>
</div>
<p>[fig:upload_inbox]</p>
<div class="figure" id="id5">
<a class="reference internal image-reference" href="_images/upload_progress.png"><img alt="Once selected from the file browser you can start the upload and observe progress in the right side pane." src="_images/upload_progress.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">Once selected from the file browser you can start the upload and
observe progress in the right side pane.</span></p>
</div>
<p>[fig:upload_progress]</p>
<p>From the inbox data needs to be submitted to the annotation pipeline.
Once files are uploaded, the inbox allows a set of operations to be
performed.</p>
</div>
<div class="section" id="expected-upload-speeds">
<h4>Expected upload speeds<a class="headerlink" href="#expected-upload-speeds" title="Permalink to this headline">¶</a></h4>
<p>Based on observed values, upload times per 1 GB (<span class="math notranslate nohighlight">\(10^9\)</span> bytes)
vary from 2 minutes to over an hour, with typical times being 10 to 15
minutes. Your experience will vary depending on the speed of your
connection to the internet and the quality of service in your region.</p>
<p>Table 1 summarizes observed upload times that might help users estimate
how long the upload should take.</p>
<table border="1" class="docutils" id="id6">
<caption><span class="caption-text">Summary of upload times</span><a class="headerlink" href="#id6" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="41%" />
<col width="24%" />
<col width="35%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Technology</th>
<th class="head">Rate (bit/s)</th>
<th class="head">Time for 1GB Upload</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Modem 14.4 (2400 baud)</td>
<td>14.4 kbit/s</td>
<td>154 hours</td>
</tr>
<tr class="row-odd"><td>ADSL Lite</td>
<td>1.5 Mbit/s</td>
<td>1.5 hours</td>
</tr>
<tr class="row-even"><td>Ethernet</td>
<td>10 Mbit/s</td>
<td>13.33 minutes</td>
</tr>
<tr class="row-odd"><td>T3</td>
<td>44.736 Mbit/s</td>
<td>3 minutes</td>
</tr>
<tr class="row-even"><td>Fast Ethernet</td>
<td>100 Mbit/s</td>
<td>1.33 minutes</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="frequent-issues-with-data-uploading">
<h4>Frequent issues with data uploading<a class="headerlink" href="#frequent-issues-with-data-uploading" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>Old browser version will not provide good throughput with the upload
and may fail to execute the Javascript for the uploader properly.
Update to the latest version of Firefox for optimal performance.</li>
<li>Browser-add-ons have in several occasions blocked uploads or led to
aborted uploads in the past. Disable those add-ons temporarily for
the duration of the upload.</li>
<li>In rare cases network devices have been presenting problems for the
upload. Some institutions have not anticipated the use of the http
protocol to transfer large data sets. In these cases the best option
is to find another network location for the transfer.</li>
</ul>
</div>
<div class="section" id="file-filters-in-place-for-uploaded-files">
<h4>File filters in place for uploaded files.<a class="headerlink" href="#file-filters-in-place-for-uploaded-files" title="Permalink to this headline">¶</a></h4>
<p>Since MG-RAST has been designed to work with metagenomic and
metatranscriptomic datasets, there is a filter in place trying to
identify datasets not suitable for MG-RAST. Those datasets will be
colored red in the inbox listing and cannot be submitted. Following are
the criteria for rejection:</p>
<ul class="simple">
<li>Protein sequences – MG-RAST is optimized to perform translation from
DNA to proteins.</li>
<li>Reads shorter than 75 basepairs – The gene prediction stage
performance deteriorates significantly with shorter reads.</li>
<li>genomes – Submissions with complete genomes or a small number of
contigs are rejected as well. Here our sister service RAST at
<a class="reference external" href="http://rast.nmpdr.org">http://rast.nmpdr.org</a> should be used instead of MG-RAST.</li>
<li>Files that are too small (sequence data less than 1 Mbp) – Files that
are too small for MG-RAST to properly function are rejected at the
submission stage. The minimal size requirement is 1 megabasepair.</li>
<li>Corrupted files – FASTA and FASTQ files which do not conform to the
format standard, e.g. if the number of unique identifiers does not
match the number of sequence records in a file, the file is
considered corrupt.</li>
<li>Alignments – We cannot identify proteins from sequences containing
alignment information.</li>
<li>Colorspace – The tool chain does not function for ABIsolid sequences
in colorspace. Please translate to standard FASTA.</li>
<li>rar compressed files and Zip files over 4 GB – We cannot decompress
these files.</li>
</ul>
<p>In addition we will filter at the upload stage any Word documents, Rich
Text Format files, and all files without the extension .fna, .fasta,
.fq, .fastq, or .sff in their name.</p>
<p><strong>Note:</strong> We recommend computing an MD5 checksum and verifying that the
checksum computed by MG-RAST is identical to the locally computed
checksum. This is the best way to ensure data integrity.</p>
<p><strong>Please note:</strong> After the actual upload is complete, the system will
compute the statistics shown in Figure
<a class="reference external" href="#fig:upload_inbox">[fig:upload_inbox]</a>. Computing this information
takes some time, so the statistics for your sequence files will not be
visible immediately after you uploaded it. If the statistics are not
displayed in a reasonable time refresh your browser page to trigger the
statistics computation.</p>
</div>
<div class="section" id="submit-data-for-processing">
<h4>Submit data for processing<a class="headerlink" href="#submit-data-for-processing" title="Permalink to this headline">¶</a></h4>
<p>In the second step, data needs to be submitted for processing. At
submission time you either add data to an existing study (or project) or
create a new study. Upon successful submission, data is removed from the
inbox. You will be notified via email once your submission has completed
processing. In addition you can monitor the progress of your submission
at the job status.</p>
<ul class="simple">
<li>All submitted data will stay private until the owner makes it public
or shares it with another user.</li>
<li>Providing metadata is required to make your data public and will
increase your priority in the queue.</li>
<li>The sooner you choose to make your data public, the higher your
priority in the queue will be.</li>
</ul>
<p>The submission step provides a visual aid to identify completed tasks
(the bars on the page are turning from blue (open) to green (done), see
Figures <a class="reference external" href="#fig:submission_open">[fig:submission_open]</a> and
<a class="reference external" href="#fig:submission_done">[fig:submission_done]</a>).</p>
<div class="figure" id="id7">
<a class="reference internal image-reference" href="_images/submission_open.png"><img alt="The submit page with none of the fields filled out." src="_images/submission_open.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">The submit page with none of the fields filled out.</span></p>
</div>
<p>[fig:submission_open]</p>
<div class="figure" id="id8">
<a class="reference internal image-reference" href="_images/submission_done.png"><img alt="The submit page with all bars in green indicating that the respective sections have been filled out." src="_images/submission_done.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">The submit page with all bars in green indicating that the respective
sections have been filled out.</span></p>
</div>
<p>[fig:submission_done]</p>
</div>
<div class="section" id="progress-monitoring">
<h4>Progress monitoring<a class="headerlink" href="#progress-monitoring" title="Permalink to this headline">¶</a></h4>
<p>Once data is submitted, you can monitor its progress.</p>
<div class="figure" id="id9">
<a class="reference internal image-reference" href="_images/submission_pipeline_view.png"><img alt="The jobs you have submitted are listed with their current status. A green dot indicates the stage has completed successfully, blue indicates that the current stage is in progress. Queued stages will produce an orange dot, green indicates a completed stage and red indicates error state. Gray dots will show for all stages waiting for other stages to complete." src="_images/submission_pipeline_view.png" style="width: 4in;" /></a>
<p class="caption"><span class="caption-text">The jobs you have submitted are listed with their current status. A
green dot indicates the stage has completed successfully, blue
indicates that the current stage is in progress. Queued stages will
produce an orange dot, green indicates a completed stage and red
indicates error state. Gray dots will show for all stages waiting for
other stages to complete.</span></p>
</div>
<p>[fig:submission_pipeline_view]</p>
<p>Depending on your priority (assigned based on available metadata and how
public your data is) your jobs will progress through the system. Jobs
that fail due to technical reasons (component failure etc.) will be
restarted by MG-RAST staff.</p>
<p>You will receive an email once a given data set has finished processing.</p>
</div>
</div>
<div class="section" id="cmd-line-uploader">
<h3>Cmd-line uploader<a class="headerlink" href="#cmd-line-uploader" title="Permalink to this headline">¶</a></h3>
<p>The following upload instructions are for all file types supported by
MG-RAST.</p>
<p>The mg-inbox command line tool allows upload of sequence and metadata
files and management of the user’s upload area, the inbox. In order to
operate on the inbox the user has to authenticate with an MG-RAST token.
The token can be retrieved from the “Account Management” –<span class="math notranslate nohighlight">\(&gt;\)</span>
“Manage personal preferences” –<span class="math notranslate nohighlight">\(&gt;\)</span> “Web Services” –<span class="math notranslate nohighlight">\(&gt;\)</span>
“authentication key” page via MG-RAST Web site.</p>
<ol class="arabic">
<li><p class="first">Make sure you have python installed on your system.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">pip</span><span class="o">.</span><span class="n">pypa</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="n">en</span><span class="o">/</span><span class="n">latest</span><span class="o">/</span><span class="n">installing</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
</li>
<li><p class="first">Go to the directory where you have your files to upload.</p>
</li>
<li><p class="first">Download the upload script</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ftp</span><span class="p">:</span><span class="o">//</span><span class="n">ftp</span><span class="o">.</span><span class="n">mg</span><span class="o">-</span><span class="n">rast</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">tools</span><span class="o">/</span><span class="n">upload</span><span class="o">/</span><span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span>
</pre></div>
</div>
</li>
<li><p class="first">Checkout the help options for “mg-inbox.py”. If you received an error
message that you are missing certain python libraries, you will need
to install them before you can run the script. To install python
libraries use pip install <span class="math notranslate nohighlight">\(&lt;\)</span>lib-name<span class="math notranslate nohighlight">\(&gt;\)</span> to install
the missing libraries. Note: The error message from running “python
mg-inbox.py –help” will provide the “lib-name” you are missing. You
may have a few libraries to install.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
</li>
<li><p class="first">First you need to login. The login option takes a user token and
writes a login file after successful login. For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="n">login</span> <span class="o">--</span><span class="n">token</span> <span class="o">&lt;</span><span class="n">myToken</span><span class="o">&gt;</span>
<span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="n">view</span> <span class="nb">all</span>
</pre></div>
</div>
</li>
<li><p class="first">You can upload a file into your inbox with</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="n">upload</span> <span class="o">&lt;</span><span class="n">path_to_file</span><span class="o">&gt;/&lt;</span><span class="n">file_name</span><span class="o">&gt;</span>
</pre></div>
</div>
</li>
<li><p class="first">If you have a compressed file to upload, supports gzip or bzip2</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="o">--</span><span class="n">gzip</span> <span class="n">upload</span> <span class="o">&lt;</span><span class="n">path_to_file</span><span class="o">&gt;/&lt;</span><span class="n">gzip_file</span><span class="o">&gt;</span>
<span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="o">--</span><span class="n">bzip2</span> <span class="n">upload</span> <span class="o">&lt;</span><span class="n">path_to_file</span><span class="o">&gt;/&lt;</span><span class="n">bzip2_file</span><span class="o">&gt;</span>
</pre></div>
</div>
</li>
<li><p class="first">If you have an archive file containing multiple files to upload,
supports: .zip, .tar, .tar.gz, .tar.bz2</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="n">upload</span><span class="o">-</span><span class="n">archive</span> <span class="o">&lt;</span><span class="n">path_to_file</span><span class="o">&gt;/&lt;</span><span class="n">archive_file</span><span class="o">&gt;</span>
</pre></div>
</div>
</li>
<li><p class="first">You can examine the content of your inbox with</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">mg</span><span class="o">-</span><span class="n">inbox</span><span class="o">.</span><span class="n">py</span> <span class="n">view</span> <span class="nb">all</span>
</pre></div>
</div>
</li>
<li><p class="first">You can submit your sequence files from the Upload page on the
MG-RAST web site (cmd-line option coming soon).</p>
</li>
</ol>
</div>
<div class="section" id="rest-api-uploader">
<h3>REST API uploader<a class="headerlink" href="#rest-api-uploader" title="Permalink to this headline">¶</a></h3>
<p>The following upload instructions are for using the MG-RAST REST API
with the curl program. In order to operate the API the user has to
authenticate with an MG-RAST token. The token can be retrieved from the
“Account Management” –<span class="math notranslate nohighlight">\(&gt;\)</span> “Manage personal preferences” –<span class="math notranslate nohighlight">\(&gt;\)</span>
“Web Services” –<span class="math notranslate nohighlight">\(&gt;\)</span> “authentication key” page via MG-RAST Web
site.</p>
<p>We strongly suggest that you use the scripts we provide, instead of the
native REST API.</p>
<ol class="arabic">
<li></li>
<li><p class="first">You can upload a file into your inbox with</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">X</span> <span class="n">POST</span> <span class="o">-</span><span class="n">H</span> <span class="s2">&quot;auth: &lt;myToken&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;upload=@&lt;path_to_file&gt;/&lt;file_name&gt;&quot;</span> <span class="s2">&quot;https://api.mg-rast.org/inbox&quot;</span>
</pre></div>
</div>
</li>
<li><p class="first">If you have a compressed file to upload, supports gzip or bzip2</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">X</span> <span class="n">POST</span> <span class="o">-</span><span class="n">H</span> <span class="s2">&quot;auth: &lt;myToken&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;upload=@&lt;path_to_file&gt;/&lt;gzip_file&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;compression=gzip&quot;</span> <span class="s2">&quot;https://api.mg-rast.org/inbox&quot;</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">X</span> <span class="n">POST</span> <span class="o">-</span><span class="n">H</span> <span class="s2">&quot;auth: &lt;myToken&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;upload=@&lt;path_to_file&gt;/&lt;gzip_file&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;compression=bzip2&quot;</span> <span class="s2">&quot;https://api.mg-rast.org/inbox&quot;</span>
</pre></div>
</div>
</li>
<li><p class="first">If you have an archive file containing multiple files to upload do
the following two steps, supports: .zip, .tar, .tar.gz, .tar.bz2</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">curl</span> <span class="o">-</span><span class="n">X</span> <span class="n">POST</span> <span class="o">-</span><span class="n">H</span> <span class="s2">&quot;auth: &lt;myToken&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;upload=@&lt;path_to_file&gt;/&lt;archive_file&gt;&quot;</span> <span class="s2">&quot;https://api.mg-rast.org/inbox&quot;</span>
<span class="mf">2.</span> <span class="n">curl</span> <span class="o">-</span><span class="n">X</span> <span class="n">POST</span> <span class="o">-</span><span class="n">H</span> <span class="s2">&quot;auth: &lt;myToken&gt;&quot;</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;format=&lt;one of: zip, tar, tar.gz, tar.bz2&gt;&quot;</span> <span class="s2">&quot;https://api.mg-rast.org/inbox/unpack/&lt;uploaded_file_id&gt;&quot;</span>
</pre></div>
</div>
</li>
</ol>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">MG-RAST</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="user_manual.html">MG-RAST user manual</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#under-the-hood-the-mg-rast-technology-platform">Under the hood: The MG-RAST technology platform</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#the-mg-rast-pipeline">The MG-RAST pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#mg-rast-data-products">MG-RAST data products</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#the-version-4-0-web-interface">The version 4.0 web interface</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#standard-operating-procedures-sops-for-mg-rast">Standard operating procedures SOPs for MG-RAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#putting-it-all-in-perspective">Putting It All in Perspective</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#the-downloadable-files-for-each-data-set">The downloadable files for each data set</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#terms-of-service">Terms of Service</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_manual.html#tools-and-data-used-by-mg-rast">Tools and data used by MG-RAST</a></li>
<li class="toctree-l1"><a class="reference internal" href="api.html">API — The MG-RAST Application Programming Interface</a></li>
<li class="toctree-l1"><a class="reference internal" href="api.html#example-scripts-using-the-mg-rast-rest-api">Example scripts using the MG-RAST REST API</a></li>
<li class="toctree-l1"><a class="reference internal" href="api.html#standard-operating-procedures-sops-for-mg-rast">Standard operating procedures SOPs for MG-RAST</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">FAQ – Frequently asked questions about MG-RAST</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#general">General</a></li>
<li class="toctree-l2"><a class="reference internal" href="#accounts">Accounts</a></li>
<li class="toctree-l2"><a class="reference internal" href="#upload-and-submission">Upload and Submission</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="api.html" title="previous chapter">API — The MG-RAST Application Programming Interface</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Folker Meyer and the MG-RAST team.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.5</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/faq.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>